### ISSUE TITLE
Feature request: combining multiple RRULEs

### ISSUE THREAD

**Issue by madrobby** [square/ruby-rrule#3]
Many calendar implementations support multiple `RRULE` properties. This is so that repetition rules that aren't possible to specify with one RRULE can be applied to an event.

For example, consider _"Payroll is due twice a month, on the last weekday on or before the 15th and on the last weekday of the month"_.

A set of `RRULE` properties would look like this:

```
RRULE:FREQ=MONTHLY;BYDAY=MO,TU,WE,TH,FR;BYMONTHDAY=13,14,15;BYSETPOS=-1
RRULE:FREQ=MONTHLY;BYDAY=MO,TU,WE,TH,FR;BYSETPOS=-1
```

It would be awesome if the library supported combining multiple RRULES and returning a enumerator with the distinct timestamps generated by them.

**Comment by mitchel456** [comment]
@madrobby Yeah, sounds like a great addition - were you thinking the API would be:

```ruby
rrule = RRule.parse([
  'RRULE:FREQ=MONTHLY;BYDAY=MO,TU,WE,TH,FR;BYMONTHDAY=13,14,15;BYSETPOS=-1',
  'RRULE:FREQ=MONTHLY;BYDAY=MO,TU,WE,TH,FR;BYSETPOS=-1'
])
```
and then you'd just call `between` as normal on `rrule`?

**Comment by madrobby** [comment]
Yes!

**Comment by mitchel456** [comment]
👍  Let me take a crack at that!

**Comment by csalvato** [comment]
@ordermuppet I'm guessing this was harder than it seemed? 😄

**Comment by newstler** [comment]
Any progress?

**Comment by TheHKF** [comment]
any progress on merging 2 RRULE?
